### ISSUE TITLE
Modernize unit tests to use LexicalComposer instead of DEPRECATED__USE{Plain/Rich}Text

### ISSUE THREAD

**Issue by zurfyx** [facebook/lexical#1268]
Like the Playground uses

**Comment by ChakshuGautam** [comment]
Hey, can you share a small context on why this change was made? I would like to take this up. Thanks.

**Comment by zurfyx** [comment]
That's awesome!

The component based approach is more flexible than hooks and enables to seamless have UI plugins on the app:

```
<LexicalComposer>
  <header>
    <ToolbarPlugin />
  </header>
  <RichTextPlugin />
  <footer>
    <FootNotePlugin />
  </footer>
```

Mainly solves two problems:
1. LexicalComposer has its own Context and the editor instance can be retrieved anywhere
2. The component based approach for plugin makes it possible to have UI based plugins that can be plugged in the very same way as non-UI based plugins (that would traditionally be hooks)

With plugin hooks alone and no context the Bold button on the toolbar would look like this:

```
function MyBoldToolbarButton(editor: LexialEditor) { // encapsulate useToolPlugin hook because it only does listener logic
  // Listener
  const [isBold] = useToolbarPlugin(editor); <-- pass editor instance wherever toolbar lives
  // UI
  return <i className={isBold && ...}>B</i>;
}
```

We used to have hooks in the past but now the ideal approach is what you see in App.jsx and Editor.jsx (hence why we marked them as DEPRECATED).

Additional note: we also can't kill these deprecated hooks yet because there's some Meta surfaces still using these but removing them completely from the open source code is already a big first step in the right direction
